examples/experimental/dagster-components/dagster_components/impls/dbt_project.py

```diff
@@ -0,0 +1,41 @@
+from dagster._core.definitions.definitions_class import Definitions
+from dagster._utils import pushd
+from dagster_dbt import DbtCliResource, DbtProject, dbt_assets
+from dagster_embedded_elt.sling.resources import AssetExecutionContext
+from pydantic import TypeAdapter
+from typing_extensions import Self
+
+from dagster_components import Component, ComponentLoadContext
+from dagster_components.core.component_decl_builder import ComponentDeclNode, YamlComponentDecl
+
+
+class DbtProjectComponent(Component):
+    params_schema = DbtCliResource
+
+    def __init__(self, dbt_resource: DbtCliResource):
+        self.dbt_resource = dbt_resource
+
+    @classmethod
+    def registered_name(cls) -> str:
+        return "dbt_project"
+
+    @classmethod
+    def from_decl_node(cls, context: ComponentLoadContext, decl_node: ComponentDeclNode) -> Self:
+        assert isinstance(decl_node, YamlComponentDecl)
+
+        # all paths should be resolved relative to the directory we're in
+        with pushd(str(decl_node.path)):
+            loaded_params = TypeAdapter(cls.params_schema).validate_python(
+                decl_node.defs_file_model.component_params
+            )
+        return cls(dbt_resource=loaded_params)
+
+    def build_defs(self, context: ComponentLoadContext) -> Definitions:
+        project = DbtProject(self.dbt_resource.project_dir)
+        project.prepare_if_dev()
```

Comment:
couldn't decide if this belonged in the `from_decl_node` bit or here, but regardless abstracting this kinda nasty bit away seems like a big win

Reply:
I think `build_defs` is the right call for now but we definitely need to formalize our component lifecycle so the answer is obvious.